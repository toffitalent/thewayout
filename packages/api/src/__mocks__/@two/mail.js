const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
