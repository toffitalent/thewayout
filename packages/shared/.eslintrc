{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
