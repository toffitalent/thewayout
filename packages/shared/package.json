{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
