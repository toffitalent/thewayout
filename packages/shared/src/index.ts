// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
