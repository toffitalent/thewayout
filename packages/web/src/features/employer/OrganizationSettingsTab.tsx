import {
  Avatar,
  Button,
  Controller,
  Field,
  Flex,
  mergeRefs,
  Text,
  Toast,
  useForm,
} from '@disruptive-labs/ui';
import { ChangeEvent, useRef } from 'react';
import {
  EmployerType,
  Industry,
  MediaType,
  NumberOfEmployers,
  states,
  YearsInBusiness,
} from '@two/shared';
import { API } from '@app/api';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { selectAuthUser } from '../auth';
import { updateEmployer } from './actions';

export const OrganizationSettingsTab = () => {
  const dispatch = useAppDispatch();
  const user = useAppSelector(selectAuthUser);
  const showError = useErrorIndicator();
  const ref = useRef<HTMLInputElement | null>(null);
  const employer = user?.employer;

  const {
    submitForm,
    reset,
    getValues,
    setValue,
    control,
    formState: { isDirty, isValid },
  } = useForm({
    defaultValues: {
      name: employer?.name || '',
      industry: employer?.industry || '',
      description: employer?.description || '',
      yearsInBusiness: employer?.yearsInBusiness || '',
      numberOfEmployees: employer?.numberOfEmployees || '',
      address: employer?.address || '',
      city: employer?.city || '',
      state: employer?.state || '',
      postalCode: employer?.postalCode || '',
      avatar: employer?.avatar || '',
      file: null as File | null,
    } as Partial<EmployerType> & { file?: File | null },
    onSubmit: async (values) => {
      if (user && user.employer) {
        const { id, ...rest } = user.employer;
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { avatar, file, ...patch } = values;

        if (file) {
          const { key } = await API.uploads.create(MediaType.Avatar, file);
          (patch as Partial<EmployerType>).avatar = key;
        }

        dispatch(updateEmployer({ id, ...rest, ...patch }))
          .unwrap()
          .catch((e) => {
            showError(e);
          })
          .then(() => {
            Toast.success('Profile updated!');
          });

        reset(getValues());
      }
    },
  });

  if (!employer) {
    return null;
  }

  const handleImgChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.target.files) {
      setValue('avatar', URL.createObjectURL(event.target.files[0]), { shouldDirty: true });
      setValue('file', event.target.files[0]);
    }
  };

  return (
    <Flex mt={8} direction="row" pb={5} container>
      <Flex item xs={12} md={4} pr={5} mb={8}>
        <Text fontSize="xl" fontWeight="700">
          Organization Information
        </Text>
      </Flex>
      <Flex item xs={12} md={8}>
        <form onSubmit={submitForm}>
          <Controller
            name="avatar"
            control={control}
            render={({ field: { ref: controllerRef } }) => (
              <Flex direction="row" mb={10}>
                <Avatar
                  size="2xl"
                  name={`${user?.employer?.name}`}
                  src={getValues('avatar') || user.employer?.avatar}
                />
                <Flex mt={5} ml={5}>
                  <Button variant="outline" mb={2} onClick={() => ref.current?.click()}>
                    Edit Picture
                  </Button>
                  <input
                    type="file"
                    accept="image/jpg, image/png, image/gif"
                    hidden
                    ref={mergeRefs(ref, controllerRef)}
                    onChange={handleImgChange}
                  />
                  <Text fontSize="sm" color="grey.600">
                    JPG, GIF or PNG. 1MB max.
                  </Text>
                </Flex>
              </Flex>
            )}
          />
          <Field
            name="name"
            type="text"
            label="Organization Name"
            validate={{ required: true }}
            control={control}
            width="1/2"
            fluid
            display="inline-block"
            pr={1}
          />
          <Field
            name="industry"
            type="select"
            options={Object.values(Industry).map((el) => ({ label: el, value: el }))}
            label="Industry"
            validate={{ required: true }}
            control={control}
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Field
            name="description"
            type="text"
            label="Organization Description"
            validate={{ required: true }}
            control={control}
            fluid
            multiline
            helperText="Max 1,000 characters."
          />
          <Field
            name="yearsInBusiness"
            type="select"
            options={Object.values(YearsInBusiness).map((el) => ({
              label: `${el} years`,
              value: el,
            }))}
            label="Years in Business"
            validate={{ required: true }}
            control={control}
            width="1/2"
            display="inline-block"
            pr={1}
          />
          <Field
            name="numberOfEmployees"
            type="select"
            options={Object.values(NumberOfEmployers).map((el) => ({ label: el, value: el }))}
            label="Number of Employees"
            validate={{ required: true }}
            control={control}
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Field
            name="address"
            type="text"
            label="Business Address"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pr={1}
          />
          <Field
            name="city"
            type="text"
            label="Town / City"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Field
            name="state"
            type="select"
            options={Object.entries(states).map(([key, value]) => ({ value: key, label: value }))}
            label="State"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pr={1}
          />
          <Field
            name="postalCode"
            type="text"
            label="Postcode / Zipcode"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Button colorScheme="primary" type="submit" mt={8} disabled={!isValid || !isDirty}>
            Save
          </Button>
        </form>
      </Flex>
    </Flex>
  );
};
