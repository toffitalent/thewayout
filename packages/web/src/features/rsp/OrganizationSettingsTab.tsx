import {
  Avatar,
  Box,
  Button,
  Controller,
  Field,
  Flex,
  mergeRefs,
  Text,
  Toast,
  useForm,
  WizardChoices,
} from '@disruptive-labs/ui';
import { ChangeEvent, ReactNode, useRef } from 'react';
import {
  JusticeStatus,
  justiceText,
  MediaType,
  offenseCategories,
  OffenseCategory,
  offenseText,
  RspType,
  states,
  Support,
  supportText,
  VeteranOrJustice,
} from '@two/shared';
import { API } from '@app/api';
import { AllowedOffenseForm } from '@app/components/CustomStepAllowedOffense';
import { PhoneNumberInput } from '@app/components/PhoneNumberInput';
import { clientProfile, ClientProfileEnums } from '@app/data/clientProfileText';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { formatPhoneNumber } from '@app/utils';
import { getWizardOffense } from '@app/utils/getWizardOffense';
import { updateRsp } from './actions';
import { ServiceAreaForm } from './CustomStepServicesArea';
import { selectRsp } from './reducer';

interface SectionProps {
  title: string;
  description?: string;
  children: ReactNode;
}

const Section = ({ title, description, children }: SectionProps) => (
  <Flex mt={8} direction="row" borderBottom pb={5}>
    <Box width="1/3" pr={5}>
      <Text fontSize="xl" fontWeight="700">
        {title}
      </Text>
      {description && (
        <Text fontSize="sm" color="grey.600">
          {description}
        </Text>
      )}
    </Box>
    <Flex width="2/3">{children}</Flex>
  </Flex>
);

export const OrganizationSettingsTab = () => {
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator({ ResourceConflict: 'Organization name already exist' });
  const rsp = useAppSelector(selectRsp);
  const ref = useRef<HTMLInputElement | null>(null);
  const isJusticeImpacted = rsp?.veteranOrJustice.includes(VeteranOrJustice.justiceImpacted);

  const {
    getValues,
    reset,
    submitForm,
    setValue,
    control,
    formState: { isDirty, isValid },
  } = useForm({
    mode: 'onChange',
    onSubmit: async (values) => {
      handleEditRsp(values);
      reset(getValues());
    },
    defaultValues: {
      avatar: rsp?.avatar || '',
      name: rsp?.name || '',
      description: rsp?.description || '',
      email: rsp?.email || '',
      phone: formatPhoneNumber(rsp?.phone || ''),
      address: rsp?.address || '',
      city: rsp?.city || '',
      state: rsp?.state || '',
      postalCode: rsp?.postalCode || '',
      file: null as File | null,
    } as Partial<RspType> & { file?: File | null },
  });

  const {
    getValues: getValuesServices,
    reset: resetServices,
    submitForm: submitFormServices,
    control: controlServices,
    formState: { isDirty: isDirtyServices, isValid: isValidServices },
  } = useForm({
    mode: 'onChange',
    onSubmit: async (values) => {
      handleEditRsp(values);
      resetServices(getValuesServices());
    },
    defaultValues: {
      support: rsp?.support,
    } as Partial<RspType>,
  });

  const {
    getValues: getValuesJusticeStatus,
    reset: resetJusticeStatus,
    submitForm: submitFormJusticeStatus,
    setValue: setValueJusticeStatus,
    control: controlJusticeStatus,
    formState: { isDirty: isDirtyJusticeStatus, isValid: isValidJusticeStatus },
  } = useForm({
    mode: 'onChange',
    onSubmit: async (values) => {
      handleEditRsp(values);
      resetJusticeStatus(getValuesJusticeStatus());
    },
    defaultValues: {
      justiceStatus: rsp?.justiceStatus || [],
    } as Partial<RspType>,
  });

  const {
    getValues: getValuesVeteranJustice,
    reset: resetVeteranJustice,
    submitForm: submitFormVeteranJustice,
    control: controlVeteranJustice,
    formState: { isDirty: isDirtyVeteranJustice, isValid: isValidVeteranJustice },
  } = useForm({
    mode: 'onChange',
    onSubmit: async (values) => {
      handleEditRsp(values);
      resetVeteranJustice(getValuesVeteranJustice());
    },
    defaultValues: {
      veteranOrJustice: rsp?.veteranOrJustice,
    } as Partial<RspType>,
  });

  const handleEditRsp = async (data: Partial<RspType> & { file?: File | null }) => {
    if (rsp) {
      const { file, ...patch } = data;
      if (file) {
        const { key } = await API.uploads.create(MediaType.Avatar, file);
        patch.avatar = key;
      }
      let clearJusticeImpactedData;
      if (
        data.veteranOrJustice !== undefined &&
        !data.veteranOrJustice.includes(VeteranOrJustice.justiceImpacted)
      ) {
        clearJusticeImpactedData = {
          justiceStatus: [],
          offenses: [],
        };
        setValueJusticeStatus('justiceStatus', []);
      }

      dispatch(
        updateRsp({
          id: rsp.id,
          ...patch,
          ...(data.phone && { phone: data.phone.replace(/[^0-9]/g, '') }),
          ...(clearJusticeImpactedData && clearJusticeImpactedData),
        }),
      )
        .unwrap()
        .catch((e) => {
          showError(e);
        })
        .then(() => {
          Toast.success('Organization updated!!');
        });
    }
  };

  const handleImgChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.target.files) {
      setValue('avatar', URL.createObjectURL(event.target.files[0]), { shouldDirty: true });
      setValue('file', event.target.files[0]);
    }
  };

  if (!rsp) {
    return null;
  }

  return (
    <>
      <Section title="Organization Information">
        <form onSubmit={submitForm}>
          <Controller
            name="avatar"
            control={control}
            render={({ field: { ref: controllerRef } }) => (
              <Flex direction="row" mb={10}>
                <Avatar size="2xl" name={rsp.name} src={getValues('avatar') || rsp.avatar} />
                <Flex mt={5} ml={5}>
                  <Button variant="outline" mb={2} onClick={() => ref.current?.click()}>
                    Edit Picture
                  </Button>
                  <input
                    type="file"
                    accept="image/jpg, image/png, image/gif"
                    hidden
                    ref={mergeRefs(ref, controllerRef)}
                    onChange={handleImgChange}
                  />
                  <Text fontSize="sm" color="grey.600">
                    JPG, GIF or PNG. 1MB max.
                  </Text>
                </Flex>
              </Flex>
            )}
          />
          <Field
            name="name"
            type="text"
            label="Organization Name"
            validate={{ required: true }}
            control={control}
            fluid
          />
          <Field
            name="description"
            type="text"
            label="Organization Description"
            validate={{ required: true }}
            control={control}
            fluid
            multiline
            helperText="Max 1,000 characters."
          />
          <Field
            name="email"
            type="text"
            label="Organization Email"
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pr={1}
            placeholder="Organization Email"
          />
          <PhoneNumberInput
            name="phone"
            label="Organization Phone"
            placeholder="Organization Phone"
            control={control}
            fluid
            maxLength={14}
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pl={1}
            setPhoneValue={(value: string) => {
              setValue('phone', value, {
                shouldDirty: true,
                shouldValidate: value.length === 14,
              });
            }}
          />
          <Field
            name="address"
            type="text"
            label="Business Address"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pr={1}
          />
          <Field
            name="city"
            type="text"
            label="Town / City"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Field
            name="state"
            type="select"
            options={Object.entries(states).map(([key, value]) => ({ value: key, label: value }))}
            label="State"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pr={1}
          />
          <Field
            name="postalCode"
            type="text"
            label="Postcode / Zipcode"
            validate={{ required: true }}
            control={control}
            fluid
            className="fieldHalfWidth"
            width="1/2"
            display="inline-block"
            pl={1}
          />
          <Button colorScheme="primary" type="submit" mt={8} disabled={!isValid || !isDirty}>
            Save
          </Button>
        </form>
      </Section>

      <Section title="Service Area">
        <Flex className="serviceAreaForm">
          <ServiceAreaForm
            data={rsp.servicesArea}
            label="Search Area"
            name="servicesArea"
            buttonLabel="Save"
            onSubmit={({ areas }) => handleEditRsp({ servicesArea: areas })}
          />
        </Flex>
      </Section>

      <Section
        title="Allowed Applicants"
        description={`We will only recommend ${rsp.name} to candidates that meet the selected attribute(s).`}
      >
        <form onSubmit={submitFormVeteranJustice}>
          <WizardChoices
            name="veteranOrJustice"
            multiple
            inline
            control={controlVeteranJustice}
            options={Object.entries(VeteranOrJustice).map(([key, value]) => ({
              label: clientProfile[key as ClientProfileEnums],
              value,
            }))}
            validate={(value) => (value.length ? undefined : 'At least one required')}
          />
          <Button
            colorScheme="primary"
            type="submit"
            mt={8}
            disabled={!isValidVeteranJustice || !isDirtyVeteranJustice}
          >
            Save
          </Button>
        </form>
      </Section>

      <Section
        title="Support Services"
        description="Please select all the support services that you provide."
      >
        <form onSubmit={submitFormServices}>
          <WizardChoices
            name="support"
            multiple
            inline
            control={controlServices}
            options={Object.entries(Support).map(([key, value]) => ({
              label: supportText[key as Support],
              value,
            }))}
            validate={(support) => (support.length ? undefined : 'At least one required')}
          />
          <Button
            colorScheme="primary"
            type="submit"
            mt={8}
            disabled={!isValidServices || !isDirtyServices}
          >
            Save
          </Button>
        </form>
      </Section>

      {isJusticeImpacted && (
        <Section title="Accepted Justice Status" description="Please select all accepted statuses.">
          <form onSubmit={submitFormJusticeStatus}>
            <WizardChoices
              name="justiceStatus"
              multiple
              inline
              control={controlJusticeStatus}
              options={Object.entries(JusticeStatus).map(([key, value]) => ({
                label: justiceText[key as JusticeStatus],
                value,
              }))}
              validate={(justiceStatus) =>
                justiceStatus.length ? undefined : 'At least one required'
              }
            />
            <Button
              colorScheme="primary"
              type="submit"
              mt={8}
              disabled={!isValidJusticeStatus || !isDirtyJusticeStatus}
            >
              Save
            </Button>
          </form>
        </Section>
      )}

      {isJusticeImpacted && (
        <Section
          title="Allowed Offenses"
          description={`We will only recommend ${rsp.name} to candidates that meet your allowable offenses (per State and Federal statutes).`}
        >
          <div className="offensesSection">
            <AllowedOffenseForm
              fields={Object.entries(offenseCategories).map(([key, values]) => ({
                name: `offense${key[0].toUpperCase() + key.slice(1)}`,
                label: values.name,
                type: 'choices',
                options: values.categories.map((el) => ({ label: offenseText[el], value: el })),
                inline: true,
                multiple: true,
                className: 'offenseCategoryStep',
                validate: (offense: OffenseCategory[]) =>
                  offense.length ? undefined : 'At least one required',
              }))}
              defaultValues={getWizardOffense(rsp?.offenses || [])}
              onSubmit={({
                offenseDrugs,
                offensePropertyDamage,
                offenseSexual,
                offenseTheft,
                offenseViolent,
                offenseWhiteCollar,
                offenseMotorVehicle,
              }) =>
                handleEditRsp({
                  offenses: [
                    ...offenseDrugs,
                    ...offensePropertyDamage,
                    ...offenseSexual,
                    ...offenseTheft,
                    ...offenseViolent,
                    ...offenseWhiteCollar,
                    ...offenseMotorVehicle,
                  ],
                })
              }
              isRequired
              shouldReset
              buttonLabel="Save"
            />
          </div>
        </Section>
      )}
    </>
  );
};
