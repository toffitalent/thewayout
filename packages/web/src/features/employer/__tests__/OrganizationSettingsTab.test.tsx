import { UserType } from '@two/shared';
import { createState, render, waitFor } from '@test';
import { OrganizationSettingsTab } from '../OrganizationSettingsTab';

const initialState = createState({
  auth: { user: { employer: { name: 'Test' }, type: UserType.Employer } },
});

describe('OrganizationSettingsTab', () => {
  test('renders correctly', async () => {
    const { container, findByText } = render(<OrganizationSettingsTab />, { initialState });
    await waitFor(async () =>
      expect(await findByText('Organization Information')).toBeInTheDocument(),
    );

    expect(container).toMatchSnapshot();
  });
});
