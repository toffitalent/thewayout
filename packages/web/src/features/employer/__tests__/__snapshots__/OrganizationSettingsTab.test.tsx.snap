// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSettingsTab renders correctly 1`] = `
<div>
  <div
    class="container flexDirection-row marginTop-8 paddingBottom-5"
  >
    <div
      class="item bp-xs-12 bp-md-4 marginBottom-8 paddingRight-5"
    >
      <p
        class="text fontSize-xl fontWeight-700"
      >
        Organization Information
      </p>
    </div>
    <div
      class="item bp-xs-12 bp-md-8"
    >
      <form>
        <div
          class="flexDirection-row marginBottom-10"
        >
          <div
            class="avatar s-2xl"
          >
            <div
              aria-label="Test"
              class="initials"
            >
              T
            </div>
          </div>
          <div
            class="marginTop-5 marginLeft-5"
          >
            <button
              class="button md outline c-primary marginBottom-2"
              type="button"
            >
              Edit Picture
            </button>
            <input
              accept="image/jpg, image/png, image/gif"
              hidden=""
              type="file"
            />
            <p
              class="text color-grey-600 fontSize-sm"
            >
              JPG, GIF or PNG. 1MB max.
            </p>
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r0:"
          >
            Organization Name
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r0:"
              name="name"
              type="text"
              value="Test"
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r1:"
          >
            Industry
          </label>
          <div
            class="root primary md fluid md"
          >
            <select
              class="input"
              id=":r1:"
              name="industry"
            >
              <option
                value="Accounting"
              >
                Accounting
              </option>
              <option
                value="Agriculture / Landscaping"
              >
                Agriculture / Landscaping
              </option>
              <option
                value="Automobile dealer"
              >
                Automobile dealer
              </option>
              <option
                value="Construction"
              >
                Construction
              </option>
              <option
                value="Consulting"
              >
                Consulting
              </option>
              <option
                value="Contracting"
              >
                Contracting
              </option>
              <option
                value="Education"
              >
                Education
              </option>
              <option
                value="Elderly Services"
              >
                Elderly Services
              </option>
              <option
                value="Engineering"
              >
                Engineering
              </option>
              <option
                value="Finance"
              >
                Finance
              </option>
              <option
                value="Government"
              >
                Government
              </option>
              <option
                value="Grocery"
              >
                Grocery
              </option>
              <option
                value="Hospitality"
              >
                Hospitality
              </option>
              <option
                value="Legal Services"
              >
                Legal Services
              </option>
              <option
                value="Manufacturing"
              >
                Manufacturing
              </option>
              <option
                value="Medical"
              >
                Medical
              </option>
              <option
                value="Merchant Wholesaler"
              >
                Merchant Wholesaler
              </option>
              <option
                value="Plumbing & HVAC"
              >
                Plumbing & HVAC
              </option>
              <option
                value="Real Estate"
              >
                Real Estate
              </option>
              <option
                value="Restaurant"
              >
                Restaurant
              </option>
              <option
                value="Retail"
              >
                Retail
              </option>
              <option
                value="Social Service"
              >
                Social Service
              </option>
              <option
                value="Technology"
              >
                Technology
              </option>
              <option
                value="Truck Transportation"
              >
                Truck Transportation
              </option>
              <option
                value="Warehousing / Storage"
              >
                Warehousing / Storage
              </option>
            </select>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="control"
          role="group"
        >
          <label
            class="label"
            for=":r2:"
          >
            Organization Description
          </label>
          <div
            class="root primary md fluid md multiline"
          >
            <textarea
              aria-describedby=":r2:-helptext"
              class="input"
              id=":r2:"
              name="description"
              rows="4"
              type="text"
            />
          </div>
          <div
            aria-live="polite"
            class="helperText"
            id=":r2:-helptext"
          >
            Max 1,000 characters.
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r3:"
          >
            Years in Business
          </label>
          <div
            class="root primary md fluid md"
          >
            <select
              class="input"
              id=":r3:"
              name="yearsInBusiness"
            >
              <option
                value="0-3"
              >
                0-3 years
              </option>
              <option
                value="3-5"
              >
                3-5 years
              </option>
              <option
                value="5-10"
              >
                5-10 years
              </option>
              <option
                value="10+"
              >
                10+ years
              </option>
            </select>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r4:"
          >
            Number of Employees
          </label>
          <div
            class="root primary md fluid md"
          >
            <select
              class="input"
              id=":r4:"
              name="numberOfEmployees"
            >
              <option
                value="1-50"
              >
                1-50
              </option>
              <option
                value="51-200"
              >
                51-200
              </option>
              <option
                value="201-500"
              >
                201-500
              </option>
              <option
                value="501-1000"
              >
                501-1000
              </option>
              <option
                value="1001-5000"
              >
                1001-5000
              </option>
              <option
                value="5001-10000"
              >
                5001-10000
              </option>
              <option
                value="10000+"
              >
                10000+
              </option>
            </select>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r5:"
          >
            Business Address
          </label>
          <div
            class="root primary md fluid md fieldHalfWidth"
          >
            <input
              class="input"
              id=":r5:"
              name="address"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r6:"
          >
            Town / City
          </label>
          <div
            class="root primary md fluid md fieldHalfWidth"
          >
            <input
              class="input"
              id=":r6:"
              name="city"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r7:"
          >
            State
          </label>
          <div
            class="root primary md fluid md fieldHalfWidth"
          >
            <select
              class="input"
              id=":r7:"
              name="state"
            >
              <option
                value="AL"
              >
                Alabama
              </option>
              <option
                value="AK"
              >
                Alaska
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="AR"
              >
                Arkansas
              </option>
              <option
                value="CA"
              >
                California
              </option>
              <option
                value="CO"
              >
                Colorado
              </option>
              <option
                value="CT"
              >
                Connecticut
              </option>
              <option
                value="DE"
              >
                Delaware
              </option>
              <option
                value="DC"
              >
                District Of Columbia
              </option>
              <option
                value="FL"
              >
                Florida
              </option>
              <option
                value="GA"
              >
                Georgia
              </option>
              <option
                value="HI"
              >
                Hawaii
              </option>
              <option
                value="ID"
              >
                Idaho
              </option>
              <option
                value="IL"
              >
                Illinois
              </option>
              <option
                value="IN"
              >
                Indiana
              </option>
              <option
                value="IA"
              >
                Iowa
              </option>
              <option
                value="KS"
              >
                Kansas
              </option>
              <option
                value="KY"
              >
                Kentucky
              </option>
              <option
                value="LA"
              >
                Louisiana
              </option>
              <option
                value="ME"
              >
                Maine
              </option>
              <option
                value="MD"
              >
                Maryland
              </option>
              <option
                value="MA"
              >
                Massachusetts
              </option>
              <option
                value="MI"
              >
                Michigan
              </option>
              <option
                value="MN"
              >
                Minnesota
              </option>
              <option
                value="MS"
              >
                Mississippi
              </option>
              <option
                value="MO"
              >
                Missouri
              </option>
              <option
                value="MT"
              >
                Montana
              </option>
              <option
                value="NE"
              >
                Nebraska
              </option>
              <option
                value="NV"
              >
                Nevada
              </option>
              <option
                value="NH"
              >
                New Hampshire
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NM"
              >
                New Mexico
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="NC"
              >
                North Carolina
              </option>
              <option
                value="ND"
              >
                North Dakota
              </option>
              <option
                value="OH"
              >
                Ohio
              </option>
              <option
                value="OK"
              >
                Oklahoma
              </option>
              <option
                value="OR"
              >
                Oregon
              </option>
              <option
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                value="RI"
              >
                Rhode Island
              </option>
              <option
                value="SC"
              >
                South Carolina
              </option>
              <option
                value="SD"
              >
                South Dakota
              </option>
              <option
                value="TN"
              >
                Tennessee
              </option>
              <option
                value="TX"
              >
                Texas
              </option>
              <option
                value="UT"
              >
                Utah
              </option>
              <option
                value="VT"
              >
                Vermont
              </option>
              <option
                value="VA"
              >
                Virginia
              </option>
              <option
                value="WA"
              >
                Washington
              </option>
              <option
                value="WV"
              >
                West Virginia
              </option>
              <option
                value="WI"
              >
                Wisconsin
              </option>
              <option
                value="WY"
              >
                Wyoming
              </option>
            </select>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12,16c-0.3,0-0.5-0.1-0.7-0.3l-6-6c-0.4-0.4-0.4-1,0-1.4s1-0.4,1.4,0l5.3,5.3l5.3-5.3c0.4-0.4,1-0.4,1.4,0s0.4,1,0,1.4l-6,6C12.5,15.9,12.3,16,12,16z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r8:"
          >
            Postcode / Zipcode
          </label>
          <div
            class="root primary md fluid md fieldHalfWidth"
          >
            <input
              class="input"
              id=":r8:"
              name="postalCode"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="button md solid c-primary disabled marginTop-8"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
